### PR TITLE
Refactoring the Form class

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
Refactoring the `Magento/Sales/Controller/Guest/Form` by removing the ObjectManager usage.

Also, this PR includes integration test coverage for the refactored form.  The integration test checks `Orders and Returns` form functionality, which should be available only for guests and correct data.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: dev/tests/integration/testsuite/Magento/Sales/Controller/Guest/FormTest.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Sales\Controller;
```
Comment (rogyar) [inline dev/tests/integration/testsuite/Magento/Sales/Controller/Guest/FormTest.php]: Please, correct the namespace. Should be `Magento\Sales\Controller\Guest`

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by eduard13** [comment]
@rogyar the mentioned point was adjusted and PR was merged.
Thank you.

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-4394](https://jira.corp.magento.com/browse/ENGCOM-4394) has been created to process this Pull Request

**Pull request merged**